backend/src/main/java/com/google/sps/servlets/LoadUserPreferencesServlet.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.sps.LoadUserPreferencesHelper;
+import com.google.sps.UsernameService;
+import com.google.sps.datastore.UserDatastore;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns a JSON object of the user's preferences */
+@WebServlet("/api/v1/load-user")
+public class LoadUserPreferencesServlet extends HttpServlet {
+
+  // Get DatastoreService and instantiate User Datastore
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final UserDatastore userDatastore = new UserDatastore(datastore);
+
+  private final UsernameService usernameService = new UsernameService();
```

Comment:
once the helper becomes a field on the servlet, these don't need to be fields anymore

Reply:
I'm not sure I understand what you mean. LoadUserPreferencesHelper takes in userDatastore and usernameService as parameters, so wouldn't this also need to be declared and initialized here but before the declaration of the helper?

Reply:
They can be created inline like this:

```
  private final LoadUserPreferencesHelper loadUserPreferencesHelper =
      new LoadUserPreferencesHelper(
          new UserDatastore(DatastoreServiceFactory.getDatastoreService()),
          new UsernameService());
```